string sentence
---
bool finish
int32 data_num